src/store/modules/product/mutations.ts

```diff
@@ -3,6 +3,12 @@ import ProductState from './ProductState'
 import * as types from './mutation-types'
 
 const mutations: MutationTree <ProductState> = {
+  [types.PRODUCT_CURRENT_UPDATED] (state, payload) {
+    state.current = payload.product
```

Comment:
Directly use payload here
```suggestion
    state.current = payload
```